File: ClientApp/src/app/interfaces/Product/UserInteract/Response/ReviewResponse.cs
namespace Contracts.Model.Product.UserInteract.Response;

public class ReviewResponse : BaseResponse
{
	public Guid Id { get; set; }
	public string? Description { get; set; }
	public float Rating { get; set; }
	public string? Title { get; set; }
	public string UserId { get; set; }
	public Guid OptionId { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/UserInteract/Response/reviewResponse.ts
import {BaseResponse} from "../../../baseResponse";
import {UsersListResponse} from "@core/interfaces/Security/Response/usersListResponse";

export interface ReviewResponse extends BaseResponse {
  id: number;
  description: string | null;
  rating: number;
  title: string | null;
  userId: string;
  optionId: string;
}

export interface DetailReviewsResponse extends ReviewResponse {
  id: number;
  description: string | null;
  rating: number;
  title: string | null;
  userId: string;
  optionId: string;
  user: UsersListResponse;
}
